sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -186,6 +187,9 @@ export class ApplicationService extends RecordDataModelService<Application> {
       application.studentAssessments = [originalAssessment];
       application.currentAssessment = originalAssessment;
       application.submittedDate = now;
+      application.location = {
```

Comment:
Not related to this PR.
We need to set the location here in two places because of the "application submit" and "application edit". Are we not supposed to do the same with the offering set at line 152? I may be missing something 😄 
@ann-aot @dheepak-aot @guru-aot thoughts?

Reply:
did you mean in the draft and submit?. the offering is to assigned to `originalAssessment` before the condition itself, right (line 163) so, it will be updated in both conditions. Not sure, if if i answered the qn or missed anything

Reply:
You are right @andrewsignori-aot. Like the way how offering is done, this assignment must happen before line 165 to make sure both scenarios Draft->Submitted and Submit->ReSubmit to be handled.

![image](https://user-images.githubusercontent.com/54600590/232160261-cc38fe50-bbbd-4d02-9953-22132e11cf36.png)

Reply:
@dheepak-aot my comment was about what @ann-aot replied. I missed the point that the `originalAssessment` is used in both scenarios.
About your comment @dheepak-aot, if I followed correctly, in case an "application edit" is happening, a new application object (`newApplication`) is created and we need to populate it also, unless I am missing something again 😉

Reply:
Oh, yes. I completely missed it. Got it now.